tests/shared/src/test/scala/scala/meta/tests/parsers/dotty/GivenUsingSuite.scala

```diff
@@ -0,0 +1,513 @@
+        List(
+          List(tparam("a", "Int")),
+          List(tparamUsing("ui", "UInt")),
+          List(tparamUsing("us", "UString"))
+        ),
+        Some(pname("Boolean")),
+        tname("???")
+      )
+    )
+    runTestAssert[Stat]("def f(a: Int)(using UInt)(using UString): Boolean = ???")(
+      Defn.Def(
+        Nil,
+        tname("f"),
+        Nil,
+        List(
+          List(tparam("a", "Int")),
+          List(tparamUsing("", "UInt")),
+          List(tparamUsing("", "UString"))
+        ),
+        Some(pname("Boolean")),
+        tname("???")
+      )
+    )
+  }
+
+  test("using-many-single-paren") {
+    val paramByName =
+      Term.Param(List(Mod.Using()), tname("us"), Some(Type.ByName(pname("UString"))), None)
+    runTestAssert[Stat]("def f(a: Int)(using ui: UInt, us: => UString): Boolean = ???")(
+      Defn.Def(
+        Nil,
+        tname("f"),
+        Nil,
+        List(List(tparam("a", "Int")), List(tparamUsing("ui", "UInt"), paramByName)),
+        Some(pname("Boolean")),
+        tname("???")
+      )
+    )
+    runTestAssert[Stat]("def f(a: Int)(using UInt, UString): Boolean = ???")(
+      Defn.Def(
+        Nil,
+        tname("f"),
+        Nil,
+        List(List(tparam("a", "Int")), List(tparamUsing("", "UInt"), tparamUsing("", "UString"))),
+        Some(pname("Boolean")),
+        tname("???")
+      )
+    )
+  }
+
+  test("using-complex") {
+    runTestAssert[Stat]("def f(x: String)(using Int)(y: String)(using b: Int): Unit = ???")(
+      Defn.Def(
+        Nil,
+        tname("f"),
+        Nil,
+        List(
+          List(tparam("x", "String")),
+          List(tparamUsing("", "Int")),
+          List(tparam("y", "String")),
+          List(tparamUsing("b", "Int"))
+        ),
+        Some(pname("Unit")),
+        tname("???")
+      )
+    )
+  }
+
+  test("using-mix-named-anonymous-error".ignore) {
+    runTestError(
+      "def f(using a: Int, String): Unit = ???",
+      "unable to mix named and anonymous using"
+    )
+    runTestError(
+      "def f(using Int, b: String): Unit = ???",
+      "unable to mix named and anonymous using"
+    )
+  }
+
+  test("using-multiple-using-single-parent-error".ignore) {
+    runTestError(
+      "def f(using a: Int, using b: String): Unit = ???",
+      "using only for first parameter"
+    )
+  }
+
+  test("using-added-middle-paren-error".ignore) {
+    runTestError("def f(a: Int, using b: String): Unit = ???", "using only for first parameter")
+  }
+
+  test("using-named-with-by-name-parameter") {
+    val usingByName =
+      Term.Param(List(Mod.Using()), tname("a"), Some(Type.ByName(pname("Int"))), None)
+    runTestAssert[Stat]("def f(using a: => Int): Unit = ???")(
+      Defn.Def(Nil, tname("f"), Nil, List(List(usingByName)), Some(pname("Unit")), tname("???"))
+    )
+  }
+
+  test("using-anonymous-with-by-name-parameter-error".ignore) {
+    runTestError("def f(using => String): Unit = ???", "anonymous using by-name invalid")
```

Comment:
Shouldn't it just be something like `an identifier expected, but eof found` <- that's what dotty suggests